source/OptimaJet.DWKit.StarterApplication/Repositories/BaseRepository.cs

```diff
@@ -40,24 +28,52 @@ public class BaseRepository<TEntity, TId> : DefaultEntityRepository<TEntity, TId
     {
         protected readonly DbSet<TEntity> dbSet;
         protected readonly CurrentUserRepository currentUserRepository;
+        protected readonly StatusUpdateService statusUpdateService;
         protected readonly DbContext dbContext;
 
         public BaseRepository(
             ILoggerFactory loggerFactory,
             IJsonApiContext jsonApiContext,
             CurrentUserRepository currentUserRepository,
+            StatusUpdateService statusUpdateService,
             IDbContextResolver contextResolver
             ) : base(loggerFactory, jsonApiContext, contextResolver)
         {
             this.dbContext = contextResolver.GetContext();
             this.dbSet = contextResolver.GetDbSet<TEntity>();
             this.currentUserRepository = currentUserRepository;
+            this.statusUpdateService = statusUpdateService;
         }
 
         public User CurrentUser {
             get {
                 return currentUserRepository.GetCurrentUser().Result;
             }
         }
+
+        public override async Task<TEntity> UpdateAsync(TId id, TEntity entity)
+        {
+            var retval = await base.UpdateAsync(id, entity);
+            statusUpdateService.OnUpdate(retval as IStatusUpdate);
+            return retval;
+        }
+
+        public override async Task<TEntity> CreateAsync(TEntity entity)
+        {
+            var retval = await base.CreateAsync(entity);
+            statusUpdateService.OnInsert(retval as IStatusUpdate);
+            return retval;
+        }
+
+        public override async Task<bool> DeleteAsync(TId id)
+        {
+            var entity = await GetAsync(id);
+            var retval = await base.DeleteAsync(id);
+            if (retval)
```

Comment:
this event could still be published without a `retval`, we have the id and we should have a T, where T is some Entity type.

what situations does deleteAsync not return something? if it doesn't we also have the original entity from getasync.

I just think it's weird to optionally use the delete hook

Reply:
Yeah, I wasn't sure what to do in this case where DeleteAsync returned false.  Did it really not get deleted?

Reply:
I feel like there should be an exception or something if something goes wrong?
would probably need to comb through the source to see what's possible

Reply:
I looked through the source and slacked you why it was like that (at least in JsonApiDotNetCore).  bool is just whether the entity id was found.  The Controller returns NotFound if false and NoContent if true.